ui/src/app/machines/views/MachineDetails/MachineLogs/DownloadMenu/DownloadMenu.test.tsx

```diff
@@ -63,6 +78,161 @@ describe("DownloadMenu", () => {
     jest.useRealTimers();
   });
 
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("is disabled if there are no downloads", () => {
+    state.scriptresult.logs = {};
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <DownloadMenu systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find("ContextualMenu").prop("toggleDisabled")).toBe(true);
+  });
+
+  it("does not display a YAML output item when it does not exist", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <DownloadMenu systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(
+      wrapper
+        .find("ContextualMenu")
+        .prop("links")
+        .some(({ children }) => children === "Machine output (YAML)")
```

Comment:
This test is pretty fragile as it depends on a string not being there so if this string gets updated the test will still pass even if the code fails.

Reply:
I've updated these to use `data-test` attributes. They will still be a bit fragile but hopefully because we test for both existing/non-existing we will catch if these change.